### PR TITLE
CI with GitHub Actions

### LINKED ISSUE #128: Try GitHub Actions
GitHub Actions is now available. Even we don't have a strong motivation to move from our current CI (CircleCI), I would like to check the features and try the new service.

https://help.github.com/en/actions

### PR DESCRIPTION
**Pull request by ynamiki**
[GitHub Actions](https://help.github.com/en/actions) is now available and now we have another alternative for CI.

This pull request activates CI with GitHub Actions. We will try this GitHub Actions in addition to the current CircleCI for some weeks then decide which one to use.

At this time, I prefer GitHub Actions:
- Set up: GitHub Action has already integrated, nothing to do. CircleCI needs some processes to integrate the service to GitHub. I spent much time to this part for the initial setup and the trouble shooting (the trigger worked only for me). Note that we don't have Admin privilege, so we need to request assistance to the LF team in some case.
- Configuration: There are small difference, but both are easy to write.
- Environment: GitHub Actions provides a VM while CircleCI provides a container to run CI. I don't find any good/bad effects from this difference.

This PR resolves #128.

### CODE REVIEW

**Comment by ynamiki** [comment]
We have an error in the GitHub Actions (and this is not occurred in CircleCI):

```
golangci-lint run ./...
usig/sgx/usig-enclave.go:260:145: SA4000: identical expressions on the left and right side of the '==' operator (staticcheck)
	sgxErr := C.call_usig_seal_key(function, enclave.enclaveID, &sealedData, &sealedDataSize)
```

https://github.com/ynamiki/minbft/commit/6149de338790296ff86d2c3ca08f54eb2ff6649d/checks?check_suite_id=328572416

I think this is a false positive case reported in https://github.com/dominikh/go-tools/issues/430 and have a relationship with Go. Currently, we use Go 1.12.7, which is provided as default in the GitHub Actions environment. This is different from the version used in CircleCI: 1.11.4. I will try update the configuration to use the same version of Go and check results.

**Comment by sergefdrv** [comment]
> I think this is a false positive case reported in [dominikh/go-tools#430](https://github.com/dominikh/go-tools/issues/430) and have a relationship with Go. Currently, we use Go 1.12.7, which is provided as default in the GitHub Actions environment. This is different from the version used in CircleCI: 1.11.4. 

This seems to be linting issue. Namely, false positive of `staticcheck`. It should be fixed in `staticcheck` as of 2019.2.1. Maybe upgrading to the latest `GolangCI-Lint` will help?

**Comment by sergefdrv** [comment]
@ynamiki Thanks for doing this :+1: The configuration looks fine. Let's fix the linting issue and try it out.

**Comment by ynamiki** [comment]
The linter error has fix by upgrading golangci-lint. Thanks for the comment.

**Comment by ynamiki** [comment]
At this time, this commit will not replace the current CI by CircleCI. Let's disscuss some weeks later whether switch to this or not after trying the new CI system.

**Review by sergefdrv — APPROVED** [review]
Look good :+1: Thanks!

We would need to update the required version of Go in our ReadMe if we decide to switch to GitHub actions.

**Review by nhoriguchi — APPROVED** [review]
Thanks, I'll merge this soon, then let's see what happens :+1:

File: .github/workflows/main.yml
Comment (nhoriguchi) [inline .github/workflows/main.yml:2]: (Sorry, I should have noticed this before pushing merge button...) if we'd like to trigger a job on creating pull request, writing like `on: [push, pull_request]` here might be better?

Comment (ynamiki) [reply .github/workflows/main.yml:2]: I think it could be enough with only `push` but I will add `pull_request` if the action is not triggered by a next new PR. Thanks.

https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows

Comment (sergefdrv) [reply .github/workflows/main.yml:2]: Apparently, Naoya was right. My latest pull request (#131) did not trigger CI action in this repository. It was only triggered in my fork.

Comment (ynamiki) [reply .github/workflows/main.yml:2]: I have updated the configuration in #133. Thanks!

**Pull request merged**
